HW04/Danila_Glazov/lib/mentor.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: false
+
+# The Mentor class is used to create objects that can interact with student objects.
+class Mentor
+  attr_reader :notification
+  attr_writer :student
+
+  def initialize(name: '', surname: '')
+    @name = name
+    @surname = surname
+    @notification = Notification.new
+    @student = nil
+  end
+
+  def add_homework(title: '', description: '', student: nil)
+    student.notification.add("#{@name} add new homework")
+    Homework.new(title: title, description: description)
+  end
+
+  def notifications
+    @notification.messages
+  end
+
+  def mark_as_read!
+    @notification.check_all!
+  end
+
+  def subscribe_to!(student)
+    self.student = student
+    student.mentor = self
+  end
+
+  def reject_to_work!(homework)
+    homework.status = 'rejected'
```

Comment:
could you add some interface to homework class? It's not a good practice to update something outside the class
```suggestion
    homework.rejected!
    
    def rejected!
      homework.status = 'rejected'
    end
```